mapit_global/tests/test_import_command.py

```diff
@@ -0,0 +1,379 @@
+from __future__ import unicode_literals
+
+from contextlib import contextmanager
+import os
+from os.path import join, dirname
+import re
+
+from django.core.management import call_command
+from django.test import TestCase
+
+from mapit.models import Area, CodeType, Country, Generation, Type
+
+from .mkdtemp_contextmanager import TemporaryDirectory
+
+
+def get_example_kml(tags, include_big_square=False, include_big_square_with_hole=False, include_small_square=False):
+    polygons = [
+        (
+            include_big_square, '''
+            <Polygon>
+              <outerBoundaryIs>
+                <LinearRing>
+                  <coordinates>0,53,0  4,53,0  4,49,0  0,49,0  0,53,0 </coordinates>
+                </LinearRing>
+              </outerBoundaryIs>
+            </Polygon>
+'''),
+        (
+            include_big_square_with_hole, '''
+            <Polygon>
+              <outerBoundaryIs>
+                <LinearRing>
+                  <coordinates>0,53,0  4,53,0  4,49,0  0,49,0  0,53,0 </coordinates>
+                </LinearRing>
+              </outerBoundaryIs>
+              <innerBoundaryIs>
+                <LinearRing>
+                  <coordinates>1,52,0  3,52,0  3,50,0  1,50,0  1,52,0 </coordinates>
+                </LinearRing>
+              </innerBoundaryIs>
+            </Polygon>
+'''),
+        (
+            include_small_square, '''
+            <Polygon>
+              <outerBoundaryIs>
+                <LinearRing>
+                  <coordinates>-3,52,0  -2,52,0  -2,51,0  -3,51,0  -3,52,0 </coordinates>
+                </LinearRing>
+              </outerBoundaryIs>
+            </Polygon>
+''')
+    ]
+    polygons_xml = ''.join(p for include, p in polygons if include)
+    tags_xml = ''.join(
+        '''
+            <Data name="{key}">
+              <value>{value}</value>
+            </Data>
+'''.format(key=k, value=v)
+        for k, v in
+        sorted(tags.items())
+    )
+    name = 'Exampleshire'
+    if 'name' in tags:
+        name = tags['name']
+    return '''<?xml version='1.0' encoding='utf-8'?>
+    <kml xmlns="http://earth.google.com/kml/2.1">
+      <Folder>
+        <name>Boundaries for {name} [relation XXXXXX]</name>
+        <Placemark>
+          <name>{name}</name>
+          <ExtendedData>
+{tags_xml}
+          </ExtendedData>
+          <MultiGeometry>
+{polygons_xml}
+          </MultiGeometry>
+        </Placemark>
+      </Folder>
+   </kml>'''.format(tags_xml=tags_xml, polygons_xml=polygons_xml, name=name)
+
+
+@contextmanager
+def example_files(type_and_file_data):
+    with TemporaryDirectory() as tmp_dir:
+        for type_code, file_data in type_and_file_data.items():
```

Comment:
You can't yield twice in a contextmanager – all the calls to this only have one key/value pair, so should they just be passed in as type_code and file_data and lose this for loop?